### PR TITLE
Add support for changing the active color scheme with an action

### LINKED ISSUE #5401: Add a shortcut action for switching to a specific color scheme
Pretty self-explanatory.
```json
{ "action": { "command": "setColorScheme", "name": "<scheme name>"}, "keys": [] }
```

```c#

    [default_interface] runtimeclass SetColorSchemeArgs : IActionArgs
    {
        string Name { get; };
    };
```

I want this for the command palette, see #5400

### PR DESCRIPTION
**Pull request by zadjii-msft**
## Summary of the Pull Request

Adds the `setColorScheme` action, to change the color scheme of the active control to one given by the `name` parameter. `name` is required. If `name` is not the name of a color scheme, the action does nothing.

## References

* Being done as a stepping stone to #6689 

## PR Checklist
* [x] Closes #5401
* [x] I work here
* [ ] Tests added/passed
* [n/a] Requires documentation to be updated

## Detailed Description of the Pull Request / Additional comments

Technically, the action is being done by changing the settings of the current `TerminalSettings` of the `TermControl`. Frankly, it should be operating on a copy of the `TermControl`'s `IControlSettings`, then updating the control's settings, or the Control should just listen for changes to it's setting's properties, and update in real time (without a manual call to `UpdateSettings`. However, both those paths are somewhere unknowable beyond #6904, so we'll just do this for now.

## Validation Steps Performed

* tested manually with a scheme that exists
* tested manually with a scheme that doesn't exist

### CODE REVIEW

File: src/cascadia/TerminalApp/AppActionHandlers.cpp [hunk 1]
```diff
@@ -272,6 +272,28 @@ namespace winrt::TerminalApp::implementation
...
+    void TerminalPage::_HandleSetColorScheme(const IInspectable& /*sender*/,
+                                             const TerminalApp::ActionEventArgs& args)
+    {
+        if (const auto& realArgs = args.ActionArgs().try_as<TerminalApp::SetColorSchemeArgs>())
+        {
+            if (auto activeTab = _GetFocusedTab())
+            {
+                if (auto activeControl = activeTab->GetActiveTerminalControl())
+                {
+                    // Theoretically, if the properties in an IControlSettings
```
Comment (miniksa) [inline src/cascadia/TerminalApp/AppActionHandlers.cpp:284]: So is this a TODO or just a musing about something that is too much work to bother doing?

File: src/cascadia/TerminalApp/ActionArgs.h [hunk 2]
```diff
@@ -330,6 +331,38 @@ namespace winrt::TerminalApp::implementation
         }
     };
 
+    struct SetColorSchemeArgs : public SetColorSchemeArgsT<SetColorSchemeArgs>
+    {
+        SetColorSchemeArgs() = default;
+        GETSET_PROPERTY(winrt::hstring, SchemeName, L"");
+
+        static constexpr std::string_view NameKey{ "name" };
```
Comment (DHowett) [inline src/cascadia/TerminalApp/ActionArgs.h:339]: should it be `colorScheme` for equivalence with the profile?

File: src/cascadia/TerminalApp/AppActionHandlers.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalApp/AppActionHandlers.cpp:284]: musing i think

File: src/cascadia/TerminalApp/ColorScheme.cpp [hunk 3]
```diff
@@ -68,7 +68,7 @@ ColorScheme::~ColorScheme()
 // - terminalSettings: the object to apply our settings to.
 // Return Value:
 // - <none>
-void ColorScheme::ApplyScheme(TerminalSettings terminalSettings) const
+void ColorScheme::ApplyScheme(winrt::Microsoft::Terminal::Settings::IControlSettings& terminalSettings) const
```
Comment (DHowett) [inline src/cascadia/TerminalApp/ColorScheme.cpp:71]: conflict w/ carlos

File: src/cascadia/TerminalApp/ColorScheme.h [hunk 4]
```diff
@@ -38,7 +38,7 @@ class TerminalApp::ColorScheme
     ColorScheme(std::wstring name, til::color defaultFg, til::color defaultBg, til::color cursorColor);
     ~ColorScheme();
 
-    void ApplyScheme(winrt::TerminalApp::TerminalSettings terminalSettings) const;
+    void ApplyScheme(winrt::Microsoft::Terminal::Settings::IControlSettings& terminalSettings) const;
```
Comment (DHowett) [inline src/cascadia/TerminalApp/ColorScheme.h:41]: conflict w/ carlos

File: src/cascadia/TerminalApp/Profile.cpp [hunk 5]
```diff
@@ -177,7 +177,8 @@ TerminalSettings Profile::CreateTerminalSettings(const std::unordered_map<std::w
         const auto found = schemes.find(_schemeName.value());
         if (found != schemes.end())
         {
-            found->second.ApplyScheme(terminalSettings);
+            IControlSettings controlSettings = terminalSettings;
+            found->second.ApplyScheme(controlSettings);
```
Comment (DHowett) [inline src/cascadia/TerminalApp/Profile.cpp:181]: This shouldn't be necessary

File: src/cascadia/TerminalApp/AppActionHandlers.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/AppActionHandlers.cpp:284]: Definitely just musing. I'll yank it.

File: src/cascadia/TerminalApp/ActionArgs.h (hunk 2 above)
Comment (zadjii-msft) [reply src/cascadia/TerminalApp/ActionArgs.h:339]: It absolutely should

**Comment by zadjii-msft** [comment]
> * [ ] What happens when the user misspells the scheme name?
>     * Right now, it looks like you go all the way down to `CascadiaSettings::ApplyColorScheme` and just don't apply the scheme.
>     * (Option 1) Should we have some sort of validation/warning? Then we prevent adding a dead keybinding. (this is my preferred route)
>     * (Option 2) Should we mark this keybinding as un-handled, and pass the key event straight to the terminal? (this might be a fine way to handle it if option 1 is too much work? But I still think we should do option 1)

I don't think the validation code can really handle that right now - I'll file a follow up for now.

I will do the "don't mark it as handled" thing _now_.

**Comment by zadjii-msft** [comment]
@carlos-zamora Docs PR: MicrosoftDocs/terminal#105

**Pull request merged**